### PR TITLE
feat: add tokio console tracing

### PR DESCRIPTION
**Pull request by Alexandcoats**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+cargo-features = ["profile-rustflags"]
+
```
Comment (Alex6323) [inline Cargo.toml:2]: Requires `nightly`. Should we add the following in `.cargo/config.toml`, and remove `console` profile for now?
```
# Enable Tokio's experimental tracing API.
[build]
rustflags = ["--cfg", "tokio_unstable"]
```

File: Cargo.toml [hunk 2]
```diff
@@ -83,6 +92,13 @@ api-explorer = [
 api-node = [
     "api",
 ]
+# To use this feature, you must also enable the unstable cfg like so:
+# > RUSTFLAGS="--cfg tokio_unstable" cargo build --features console
+# > cargo build --profile console --features console
+console = [
+    "dep:console-subscriber",
+    "tokio/tracing",
+]
```
Comment (Alex6323) [inline Cargo.toml:101]: Maybe I am doing something stupid, but using the `2nd` option doesn't compile for me when it tries to compile `tokio-1.18.0` with error `use of undeclared crate or module tracing`.

**Review by Alex6323 — APPROVED** [review]

File: .cargo/config.toml [hunk 3]
```diff
@@ -14,3 +14,7 @@ ci-udeps = "udeps --all-targets --all-features --backend=depinfo"
 fix-clippy = "clippy --all-targets --all-features --fix"
 fix-fmt = "fmt --all"
 fix-toml = "sort --grouped"
+
+# Enable Tokio's experimental tracing API.
+[build]
+rustflags = ["--cfg", "tokio_unstable"]
```
Comment (grtlr) [inline .cargo/config.toml:20]:
```suggestion
rustflags = ["--cfg", "tokio_unstable"]

```

File: src/runtime/scope.rs [hunk 4]
```diff
@@ -193,7 +194,8 @@ impl RuntimeScope {
     {
         let SpawnConfig { mut actor, config } = actor.into();
         let (handle, mut cx, abort_reg) = self.common_spawn(&actor, config).await;
-        let child_task = tokio::spawn(async move {
+
```
Comment (grtlr) [inline src/runtime/scope.rs:197]:
```suggestion
```

File: src/runtime/scope.rs [hunk 5]
```diff
@@ -225,6 +227,7 @@ impl RuntimeScope {
                 }
             }
         });
+
```
Comment (grtlr) [inline src/runtime/scope.rs:230]:
```suggestion
```

File: src/runtime/actor/util.rs [hunk 6]
```diff
@@ -42,7 +42,7 @@ where
         _state: &mut Self::State,
     ) -> Result<(), Self::Error> {
         let handle = cx.handle().clone();
-        tokio::spawn(async move {
+        spawn_task("Delay Event Sleeper", async move {
```
Comment (grtlr) [inline src/runtime/actor/util.rs:45]:
```suggestion
        spawn_task("delay event sleeper", async move {
```

File: src/runtime/mod.rs [hunk 7]
```diff
@@ -60,3 +60,16 @@ impl Runtime {
         }
     }
 }
+
+/// Spawn a tokio task. The provided name will be used to configure the task if console tracing is enabled.
+pub fn spawn_task<F>(name: impl AsRef<str>, task: F) -> tokio::task::JoinHandle<F::Output>
```
Comment (grtlr) [inline src/runtime/mod.rs:65]: Why not just `&str`?

File: bin/inx-chronicle/src/api/mod.rs [hunk 8]
```diff
@@ -57,7 +60,7 @@ impl Actor for ApiWorker {
         log::info!("Starting API server");
         let db = self.db.clone();
         let api_handle = cx.handle().clone();
-        let join_handle = tokio::spawn(async move {
+        let join_handle = spawn_task("Axum Server", async move {
```
Comment (grtlr) [inline bin/inx-chronicle/src/api/mod.rs:63]:
```suggestion
        let join_handle = spawn_task("Axum server", async move {
```

File: src/runtime/mod.rs (hunk 7 above)
Comment (Alexandcoats) [reply src/runtime/mod.rs:65]: 🤔 ok

**Review by grtlr — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

File: bin/inx-chronicle/src/api/mod.rs [hunk 9]
```diff
@@ -69,9 +69,28 @@ impl Actor for ApiWorker {
         let (sender, receiver) = oneshot::channel();
         log::info!("Starting API server");
         let api_handle = cx.handle().clone();
+        let port = self.config.port;
```
Comment (grtlr) [inline bin/inx-chronicle/src/api/mod.rs:72]:
```suggestion
```

File: bin/inx-chronicle/src/api/mod.rs [hunk 10]
```diff
@@ -69,9 +69,28 @@ impl Actor for ApiWorker {
...
+                            .unwrap_or_else(AllowOrigin::any),
+                    )
+                    .allow_methods(vec![Method::GET, Method::OPTIONS])
+                    .allow_headers(Any)
+                    .allow_credentials(false),
+            );
         let join_handle = spawn_task("Axum server", async move {
-            let res = Server::bind(&([0, 0, 0, 0], 9092).into())
-                .serve(routes(db).into_make_service())
+            let res = Server::bind(&([0, 0, 0, 0], port).into())
```
[the remark this answers is not in the document]
Comment (Alexandcoats) [reply bin/inx-chronicle/src/api/mod.rs:92]: No can do, as this would move `self` into the task.

[the remark this answers is not in the document]
Comment (grtlr) [reply bin/inx-chronicle/src/api/mod.rs:92]: Ah, you're right.

**Review by grtlr — APPROVED** [review]

**Review by grtlr — APPROVED** [review]

**Pull request merged**